src/external_printer.rs

```diff
@@ -0,0 +1,43 @@
+#[cfg(feature = "external_printer")]
+use crossbeam::channel::{bounded, Receiver, Sender};
+
+/// An ExternalPrinter allows to print messages of text while editing a line.
+/// The message is printed as a new line, the line-edit will continue below the
+/// output.
+#[cfg(feature = "external_printer")]
+#[derive(Debug, Clone)]
+pub struct ExternalPrinter {
+    sender: Sender<String>,
+    receiver: Receiver<String>,
+}
+
+#[cfg(feature = "external_printer")]
+impl ExternalPrinter {
+    /// Creates an ExternalPrinter to store lines with a max_cap
+    pub fn new(max_cap: usize) -> Self {
+        let (sender, receiver) = bounded::<String>(max_cap);
+        Self { sender, receiver }
+    }
+    /// Gets a Sender to use the printer externally by sending lines to it
+    pub fn sender(&self) -> Sender<String> {
+        self.sender.clone()
+    }
```

Comment:
Why have this and the explicit `print()` at the same time?

Reply:
I was thinking of print() as a simple to use convenience method. To have more control on the Sender-side I think it is useful to have a reference of the Sender<String> to make the sending more customizable (timeouts, blocking, etc.).